pkg/cmd/chat/channel/channel.go

```diff
@@ -318,3 +325,240 @@ func listCmd() *cobra.Command {
+			if err != nil {
+				return err
+			}
+
+			chType, _ := cmd.Flags().GetString("type")
+			chID, _ := cmd.Flags().GetString("id")
+
+			_, err = c.Channel(chType, chID).RemoveMembers(cmd.Context(), args, nil)
+			if err != nil {
+				return err
+			}
+
+			cmd.Println("Successfully removed user(s) from channel")
+			return nil
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.StringP("type", "t", "", "[required] Channel type such as 'messaging' or 'livestream'")
+	fl.StringP("id", "i", "", "[required] Channel id")
+	cmd.MarkFlagRequired("type")
+	cmd.MarkFlagRequired("id")
+
+	return cmd
+}
+
+func promoteModeratorCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "promote-moderators --type [channel-type] --id [channel-id] [user-id-1] [user-id-2] ...",
+		Short: "Promote users to channel moderator role",
+		Args:  cobra.MinimumNArgs(1),
+		Example: heredoc.Doc(`
+			# Promote 4 users to moderator
+			$ stream-cli chat promote-moderators --type messaging --id red-team joe mike jane jill
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := config.GetConfig(cmd).GetClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			chType, _ := cmd.Flags().GetString("type")
+			chID, _ := cmd.Flags().GetString("id")
+
+			_, err = c.Channel(chType, chID).AddModerators(cmd.Context(), args...)
+			if err != nil {
+				return err
+			}
+
+			cmd.Println("Successfully promoted users to moderators")
+			return nil
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.StringP("type", "t", "", "[required] Channel type such as 'messaging' or 'livestream'")
+	fl.StringP("id", "i", "", "[required] Channel id")
+	cmd.MarkFlagRequired("type")
+	cmd.MarkFlagRequired("id")
+
+	return cmd
+}
+
+func demoteModeratorCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "demote-moderators --type [channel-type] --id [channel-id] [user-id-1] [user-id-2] ...",
+		Short: "Demote users from moderator role",
+		Args:  cobra.MinimumNArgs(1),
+		Example: heredoc.Doc(`
+			# Demote 4 users from moderator role
+			$ stream-cli chat demote-moderators --type messaging --id red-team joe mike jane jill
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := config.GetConfig(cmd).GetClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			chType, _ := cmd.Flags().GetString("type")
+			chID, _ := cmd.Flags().GetString("id")
+
+			_, err = c.Channel(chType, chID).DemoteModerators(cmd.Context(), args...)
+			if err != nil {
+				return err
+			}
+
+			cmd.Println("Successfully demoted users from moderators")
+			return nil
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.StringP("type", "t", "", "[required] Channel type such as 'messaging' or 'livestream'")
+	fl.StringP("id", "i", "", "[required] Channel id")
+	cmd.MarkFlagRequired("type")
+	cmd.MarkFlagRequired("id")
+
+	return cmd
+}
+
+func hideCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "hide-channel --type [channel-type] --id [channel-id] --user-id [user-id]",
```

Comment:
i think we should be consistent on how we are passing users id, in commands above is treated as args and here flag.

Reply:
@yaziine that's because you can pass in multiple users to the above command.

```
add-members --type messaging --id redteam jack jill mike yassine peter
```
vs
```
hide-channel --type messaging--id redteam --user-id joe
```

I think it makes sense

Reply:
I think it's ok to have --user-id. In this case it identifies me as command executor. So this param could be named as --run-as=joe

Reply:
possibly, I would need to rename a bunch of parameters though :)

Reply:
After discussion, we agreed to go that way. All good then.